src/Agent.Sdk/SecretMasking/LegacyLoggedSecretMasker.cs

```diff
@@ -129,13 +137,29 @@ public ISecretMasker Clone()
         return new LegacyLoggedSecretMasker(this._secretMasker.Clone());
     }
 
+
+    public double ElapsedMaskingTime => this.elapsedMaskingTime;
+
     public string MaskSecrets(string input)
     {
-        return this._secretMasker.MaskSecrets(input);
+        var stopwatch = Stopwatch.StartNew();
+        string result = this._secretMasker.MaskSecrets(input);
+        Interlocked.Exchange(ref elapsedMaskingTime, stopwatch.ElapsedTicks);
```

Comment:
>Exchange [](http://example.com/codeflow?start=20&length=8)

I'm not sure what threading is in play in the agent. The masker is thread-safe for the masking operation, so I used `Interlocked.Exchange` to update the elapsed time property.